### ISSUE TITLE
Question: is it possible to extract the enumerated color space from an image?

### ISSUE THREAD

**Issue by henryborchers** [Exiv2/exiv2#386]
I'm trying to locate the enumerated color space value in jp2 files. I cannot find a way to do this for jp2 or any other image formats in Exiv2's API documentation. Can Exiv2 currently locate this information?

**Comment by clanmills** [comment]
There is no API to explicitly extract the color space of an image.  Exiv2 enumerates metadata encoded as Exif, IPTC and XMP.  If the color space is encoded in the Exif, IPTC or XMP metadata, it will be presented.  In v0.26, I added support to extract the ICC profile for several file formats (TIFF, JPEG, PNG, JP2 and WebP, I think).  

I wasn't aware that JP2 can have multiple color spaces.

We don't support any multiple page formats such as PDF which could use a different color space on different pages.

**Comment by clanmills** [comment]
I've had a private discussion with Henry.  We talked about using OpenJPEG and debugging the command: `$ exiv2 -pR image-file` to discover how Exiv2 parses the JP2 Headers (without decoding the image).  I've also suggested asking his question on the JP2 forum:  https://groups.google.com/forum/#!forum/openjpeg

I'm going to close this issue as I don't envision further work on this topic.  @henryborchers you are welcome to continue the discussion and I will reopen the topic.
